include/zephyr/rtio/rtio.h

```diff
@@ -926,15 +907,12 @@ static inline int z_impl_rtio_cqe_get_mempool_buffer(const struct rtio *r, struc
 						     uint8_t **buff, uint32_t *buff_len)
 {
 #ifdef CONFIG_RTIO_SYS_MEM_BLOCKS
-	if (cqe->flags & RTIO_CQE_FLAG_MEMPOOL_BUFFER) {
-		int map_idx = RTIO_CQE_FLAG_GET_VALUE(cqe->flags);
-		struct rtio_mempool_map_entry *entry = &r->mempool_map[map_idx];
+	if (FIELD_GET(GENMASK(7, 0), cqe->flags) == RTIO_CQE_FLAG_MEMPOOL_BUFFER) {
+		int blk_idx = FIELD_GET(GENMASK(19, 8), cqe->flags);
+		int blk_count = FIELD_GET(GENMASK(31, 20), cqe->flags);
```

Comment:
Might be worthwhile to create macros for the bitmasks of where `blk_idx` and `blk_count` exist in the flags. We seem to use this mask above lines 879-881.

Reply:
Done